### DISCUSSION TITLE
Support Gradle Configuration Cache

### DISCUSSION THREAD

**Discussion by francescocervone** [dipien/releases-hub-gradle-plugin#152] [General]
**Is your feature request related to a problem? Please describe.**
The plugin doesn't support [Gradle Configuration Cache](https://docs.gradle.org/current/userguide/configuration_cache.html). If a client wants to enable the configuration cache, the build will fail.

**Describe the solution you'd like**
Support Gradle Configuration Cache.

**Describe alternatives you've considered**
You can also declare all the tasks as not compatible with configuration cache while planning a migration ([docs](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:task_opt_out)). This way, your plugin won't make clients build fail if they enable configuration cache for now. Indeed, the default behavior makes the build fail even if there is just one problem ([docs](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:usage:ignore_problems)).

**Additional context**
Here is an example of configuration cache report after running `listDependenciesToUpgrade`. Also the `upgradeDependencies` task has issue. I haven't tested the other ones.

![Schermata 2022-09-30 alle 10 36 30](https://user-images.githubusercontent.com/6197796/193229003-8c93f0ff-5cdb-4922-8468-b66a0df298f5.png)

**Comment by maxirosson** [comment]
Hi. Thanks for reporting this. I released [v3.2.0](https://github.com/dipien/releases-hub-gradle-plugin/releases/tag/v3.2.0) with the fixes to make the tasks not compatible with configuration cache.

**Comment by maxirosson** [comment]
All the details here: https://blog.dipien.com/releases-hub-gradle-plugin-v3-2-0-c7ee5100f5bc
